providers/packetframe/api.go

```diff
@@ -0,0 +1,188 @@
+package packetframe
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+)
+
+const (
+	mediaType      = "application/json"
+	defaultBaseURL = "https://packetframe.com/api/"
+)
+
+func (c *packetframeProvider) fetchDomainList() error {
+	c.domainIndex = map[string]zone{}
+	dr := &domainResponse{}
+	endpoint := "dns/zones"
+	if err := c.get(endpoint, dr); err != nil {
+		return fmt.Errorf("failed fetching domain list (Packetframe): %s", err)
+	}
+	for _, zone := range dr.Data.Zones {
+		c.domainIndex[zone.Zone] = zone
+	}
+
+	return nil
+}
+
+func (c *packetframeProvider) getRecords(zoneID string) ([]domainRecord, error) {
+	var records []domainRecord
+	dr := &recordResponse{}
+	endpoint := fmt.Sprintf("dns/records/%s", zoneID)
+	if err := c.get(endpoint, dr); err != nil {
+		return records, fmt.Errorf("failed fetching domain list (Packetframe): %s", err)
+	}
+	records = append(records, dr.Data.Records...)
+
+	return records, nil
+}
+
+func (c *packetframeProvider) createRecord(rec *domainRecord) (*domainRecord, error) {
+	endpoint := "dns/records"
+
+	req, err := c.newRequest(http.MethodPost, endpoint, rec)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = c.client.Do(req)
+	if err != nil {
+		fmt.Println("ERROR")
```

Comment:
Is this `fmt.Println` a debug statement that should be removed?

Reply:
Whoops! I can't believe that got left in, good catch